### DISCUSSION TITLE
How to change the size of favicon?

### DISCUSSION THREAD

**Discussion by ricolxwz** [squidfunk/mkdocs-material#7490] [General]
How to change the size of favicon in the browser tab?

**Comment by alexvoss** [comment]
I was not aware that this is even possible but [it seems to be](https://stackoverflow.com/questions/59473506/changing-favicon-size-in-html#59473660), though I am not sure you will get consistent results across browsers. To add the necessary extra header information, you could [override the `extrahead` block](https://squidfunk.github.io/mkdocs-material/customization/?h=extra+head#overriding-blocks).

Would be good if you could post a screenshot of your results, showing what the use of different favicon sizes achieves. I assume it is a better rendering of an icon for high resolution screens rather than an icon that is actually larger? Having looked at a random website there is also an `apple-touch-icon`?

**Reply by ricolxwz** [reply]
Thanks!
![image](https://github.com/user-attachments/assets/1e742b04-42f3-4c3f-9060-7ba00b52b845)
Actually, it's not that small but compared to github, that's small.

**Reply by alexvoss** [reply]
Do you intend to use the Material icon? You know that [you can configure your own favicon](https://squidfunk.github.io/mkdocs-material/setup/changing-the-logo-and-icons), right?

**Reply by ricolxwz** [reply]
My icon is this one: 
![image](https://github.com/user-attachments/assets/9264a8fc-dcf4-4616-b690-ee9bed84e5b7)
Link to my page: [https://ml.ricolxwz.de](https://ml.ricolxwz.de)

**Comment by kamilkrzyskow** [comment]
Hello @ricolxwz,
You have to remember that images are squares / rectangles, and the circle is an illusion made by making the corners transparent.

Compare all of the images in some image editor like https://www.photopea.com/:
- https://github.com/favicon.ico
- https://squidfunk.github.io/mkdocs-material/assets/favicon.png
- https://ml.ricolxwz.de/assets/favicon-white-background.svg
- https://npe.cm/favicon.svg

You'll notice that GitHub's leaves almost no gaps at the edges of the square, while the theme's favicon has both margin gaps at the edges and the Material "M" logo is in a circle that too has margins, which leads to this effect on the browser tab that makes it look small in comparison.

---

Over at Nype - https://npe.cm/ due to a domain / website migration from WordPress to MkDocs there was also a change to the favicon paths. This resulted in Google search results not detecting the favicon icon due to a 404 error from the migrated site, and [due to slow indexing](https://github.com/squidfunk/mkdocs-material/discussions/7478#discussioncomment-10443084) there was no icon for a couple of weeks.

Because of that I went on to find resources on how to optimize favicons. I found this article to be helpful in general together with the website it advertised, but I have no affiliation to any of them :v::
- Definitive edition of "How to Favicon" - https://dev.to/masakudamatsu/favicon-nightmare-how-to-maintain-sanity-3al7
- Favicon Generator and Checker - https://realfavicongenerator.net/

> [!NOTE]
> Currently Material for MkDocs doesn't support setting the `sizes=` attribute for the `favicon.ico` file to enforce the hack.
> https://github.com/squidfunk/mkdocs-material/blob/6f3c05b6fa7e42279ea7604aedc4e4333131fe57/src/templates/base.html#L79-L80
> So personally, I'm just ignoring the `sizes=` attribute and use both a 48x48 ICO and SVG, and as Alex has said above, you can use customization with the `extrahead` block or `site_meta` and `super()` to move the fetch request for the icon before other assets in the `<head>` :v:

Only the `rel="icon"` meta tags are important on most "normal" websites. If you think your users will want to make a pin of your website on their smartphone screen then you can go the extra mile and add the apple-touch-icon / manifest with android images. It wasn't much more work, so Nype uses both of them, only Windows App Store icons got ignored. Something like [PWAs](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps) would be more likely to want to use those.

Also in a previous discussion it was concluded that the favicon.ico doesn't have to be at the root of the server, just anywhere on the site and the `rel="icon"` will properly direct browsers to it:
- https://github.com/squidfunk/mkdocs-material/discussions/7114

However, compare for example those links:
- https://ml.ricolxwz.de/search/search_index.json
- https://npe.cm/search/search_index.json

Both are text files without HTML with the `rel="icon"`, so browsers default to the base of the domain to fetch the favicon.
This isn't all that important, as most served filed will be HTML, but worth noting nonetheless.

**Reply by ricolxwz** [reply]
Fantastic work, thank you!
